### ISSUE TITLE
Handle scenarios when budget exceeded

### ISSUE THREAD

**Issue by raducristianpopa** [interledger/web-monetization-extension#144]
## User has a one time amount

If we reached the grant limit - we cannot create outgoing payments anymore, we display a screen in the popup (inactive icon + orange dot - user interaction) where the user will have the options to:
- Set a new one time amount: we create a new grant for the amount that the user is going to specify and revoke the previous grant
- Set a recurring amount: we create a new grant for the amount that the user is going to specify with a recurring interval (30d) and revoke the previous grant
When the user runs out of funds we need to update the extension state (new property in the storage - out_of_funds

## User has a recurring amount

If we reached the grant limit, we display the same screen as in the one time amount scenario (same options + inactive icon - orange dot)

- If the user sets a new recurring amount, we revoke the previous grant and create a new one with the specified amount
- If the user wants to set a one time amount, we need to keep the previous grant, create a new one for the specified amount and use the one time grant for all future payments until the recurring amount is available again
- When the recurring amount is available again we need to sum up the remaining amounts and we use recurring one until it runs out of funds again - no need to show the out_of_funds screen if we still have funds in the one time amount grant.

TODO:
- [x] https://github.com/interledger/web-monetization-extension/issues/376
- [x] https://github.com/interledger/web-monetization-extension/issues/425
- [x] https://github.com/interledger/web-monetization-extension/issues/377

Out of scope for [beta](https://github.com/interledger/web-monetization-extension/milestone/2) release:
- https://github.com/interledger/web-monetization-extension/issues/444
- https://github.com/interledger/web-monetization-extension/issues/445
- https://github.com/interledger/web-monetization-extension/issues/446

**Comment by sidvishnoi** [comment]
Notes: 
- https://interledgerfoundation.slack.com/archives/C054B631B97/p1718799633709149

**Comment by tselit** [comment]
Blocked by #377  and the OP depedency (@tselit to update with outlook for designs after chatting to Madalina)
